src/Button.php

```diff
@@ -25,6 +27,18 @@ public function renderView()
             $this->addClass('icon');
         }
 
+        if ($this->rightIcon && !is_object($this->rightIcon)) {
+            $this->rightIcon = $this->add(new Icon($this->rightIcon), 'Content');
+
+            if ($this->content) {
```

Comment:
that's currently incosistent in the project but i've addressed that in the next PR, but i'm not fully sure.